src/app/static/src/app/components/downloadResults/UploadModal.vue

```diff
@@ -100,10 +131,54 @@
                 "uploadFilesToADR"
             ),
             confirmUpload() {
-                const uploadFilesPayload: UploadFile[] = []
-                this.uploadFilesToAdr.forEach(value => uploadFilesPayload.push(this.uploadFiles[value]))
-                this.uploadFilesToADRAction(uploadFilesPayload);
-                this.$emit("close")
+                this.uploadFilesToAdr.forEach(value => this.uploadFilesPayload.push(this.uploadFiles[value]));
+                const readyForUpload = this.prepareFilesForUpload();
+
+                if (readyForUpload) {
+                    this.sendUploadFilesToADR();
+                }
+            },
+            sendUploadFilesToADR() {
+                this.uploadFilesToADRAction(this.uploadFilesPayload);
+                this.uploadFilesPayload = [];
+
+                this.$emit("close");
+            },
+            prepareFilesForUpload() {
+                const {summary, spectrum} = this.findSelectedUploadFiles();
+                if (summary) {
+                    this.getSummaryDownload();
+                }
+                if (spectrum) {
+                    this.getSpectrumDownload();
+                }
+
+                return this.downloadIsReady();
+            },
+            findSelectedUploadFiles() {
+                const summary = this.uploadFilesPayload.find(upload => upload.resourceType === this.outputSummary);
+                const spectrum = this.uploadFilesPayload.find(upload => upload.resourceType === this.outputSpectrum);
+
+                return {summary, spectrum}
+            },
+            downloadIsReady() {
+                const {summary, spectrum} = this.findSelectedUploadFiles();
+
+                if (summary && spectrum) {
+                    return !!this.summary.complete && !!this.spectrum.complete;
+                } else {
+                    return summary ? !!this.summary.complete : !!this.spectrum.complete;
+                }
+            },
```

Comment:
I think you might not have seen my last comment as github can sometimes hide replies to old conversations, so staring a fresh one:
>Unfortunately this suggestion is not going to work in this case but I have refactored the code to be clearer.

If I have got it a bit wrong, would it still be possible to rework your code to be more in this style, as both Mark and I seem to feel that it makes the logic more visible than multi-line with if statements, even after refactoring.
Which case are you thinking of that this suggestion would not work for? Is it the case where neither summary nor spectrum are checked ? That should return false because downloadIsReady should be false if neither are ticked? Could add a clause to check at least one of the two is true?:
`return (summary || spectrum) && (!summary || !!this.summary.complete) && (!spectrum || !!this.spectrum.complete);`

Reply:
I don't see how this code is clearer than the above.

Reply:
Obviously a degree of subjectivity here, but trying to explain why I find Emma's suggestion preferable:
- It's shorter. Not always the best metric, but in this case it's proportional to how long it took me to understand!
- It's symmetric i.e. it doesn't (slightly misleadingly) treat summary and spectrum differently
- It feels like it has less "branches", again meaning it's easier to understand the intention of the code

More generally, not sure why we need the `!!`... isn't `complete` a boolean anyway?

Reply:
Yes, to me it feels that expressing the logic as a boolean structure rather than branching ifs more transparent as well as terser - it's the structure that my brain tries to construct when looking at the branching ifs anyway to understand the meaning of what's being expressed. But agreed, it is quite subjective.

Reply:
> More generally, not sure why we need the !!... isn't complete a boolean anyway?

It is, so no we shouldn't need that!

Reply:
> > More generally, not sure why we need the !!... isn't complete a boolean anyway?
> 
> It is, so no we shouldn't need that!

!! is needed because I am getting them through Partial<>